api/liboni/onidefs.h

```diff
@@ -91,4 +91,12 @@ typedef uint64_t oni_fifo_time_t; // FIFO bound timers use 64-bit words; // TODO
 // Register size
 #define ONI_REGSZ sizeof(oni_reg_val_t)
 
+typedef struct {
+    const char *name;
+    const int major;
+    const int minor;
+    const int patch;
+    const char *preRelase;
```

Comment:
Should this just be a const int, 0 or 1? What happens if somebody tries to de-reference a const char* initialized to NULL?

Reply:
I guess there is a hint that its a char * and this is always an issue with them. And I guess you want to be able to put strings in there like Semantic versioning specifies.

Reply:
Yeah... The same happens with the driver name, actually.
We could make it so no-prelease is specified as `pre_release = ""` but we cannot actually force that beyond maybe a hacky assertion in the getter function.

As you say, that is the deal with any pointer type in C, they can always be null.